squire_lib/src/ffi_tournament.rs

```diff
@@ -53,41 +57,75 @@ impl TournamentId {
         return ptr;
     }
 
+    /// Adds a player to a tournament
+    /// On error a NULL UUID is returned
+    #[no_mangle]
+    pub unsafe extern "C" fn tid_add_player(self: Self, __name: *const c_char) -> PlayerId {
+        let name: &str = unsafe { CStr::from_ptr(__name).to_str().unwrap() };
+        let op: TournOp =
+            TournOp::RegisterPlayer(SquireAccount::new(name.to_string(), name.to_string()));
+
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get_mut(&self) {
```

Comment:
You actually don't need this first `get_mut()`. `DashMap` is a concurrent map, so it handles interior mutability for us.
```suggestion
        match FFI_TOURNAMENT_REGISTRY.get().unwrap().get_mut(&self) {
```